### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by AdamTumi**
New scheme for CODEOWNERS file.

### CODE REVIEW

**Review by jane-intel — COMMENTED** [review]
Hello! 
Your branch is hanging in the main repository of OV -- where we only store release branches. Could you please merge it and delete the branch? Proper way of doing it is to have fork and create branches there and then pull changes from your fork and branch to OV
image

**Comment by ilyachur** [comment]
@AdamTumi As I see at the current moment CODEOWNERS file contains some errors:
Screenshot 2022-08-24 at 14 12 08
Can we solve these errors before the merge?

File: CODEOWNERS [hunk 1]
```diff
@@ -29,64 +29,74 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
 # OpenVINO Auto (MULTI) plugin:
 /src/plugins/auto/  @openvinotoolkit/openvino-ie-auto-multi-maintainers
-/src/inference/include/ie/multi-device/  @openvinotoolkit/openvino-ie-auto-multi-maintainers
 
-# OpenVINO Paddle frontend:
+# OpenVINO Auto (Batch) plugin:
+/src/plugins/auto_batch/ @openvinotoolkit/openvino-hetero-maintainers
+
+# OpenVINO Hetero plugin:
+/src/plugins/hetero/ @openvinotoolkit/openvino-auto-batch-maintainers
```
[the remark this answers is not in the document]
Comment (AdamTumi) [reply CODEOWNERS]: Yes, thanks for spotting this. Next time I'll follow the rules for PR creation.

File: CODEOWNERS [hunk 2]
```diff
@@ -29,64 +29,74 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
 # OpenVINO Tests:
 /src/tests/  @openvinotoolkit/openvino-ie-tests-maintainers @openvinotoolkit/openvino-ie-test-developers
 /src/tests_deprecated/  @openvinotoolkit/openvino-ie-tests-maintainers @openvinotoolkit/openvino-ie-test-developers
-/src/tests/functional/inference_engine/ngraph_reader/  @openvinotoolkit/openvino-ie-tests-maintainers @openvinotoolkit/openvino-ngraph-maintainers
-/src/tests/functional/inference_engine/transformations/  @openvinotoolkit/openvino-ie-tests-maintainers @openvinotoolkit/openvino-ngraph-maintainers
+/src/frontends/tests/frontend/shared/ @openvinotoolkit/openvino-ie-tests-maintainers
 
 # Documentation:
 /docs/  @openvinotoolkit/openvino-docs-maintainers
-/docs/template_plugin/  @openvinotoolkit/openvino-ie-template-maintainers
```
Comment (ilyachur) [inline CODEOWNERS:88]: I think we still need to have `/docs/template_plugin/` and also `/docs/template_extension/`

File: CODEOWNERS [hunk 3]
```diff
@@ -29,64 +29,74 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
-/src/plugins/intel_cpu/thirdparty/onednn/  @openvinotoolkit/openvino-ie-cpu-maintainers @openvinotoolkit/openvino-ie-cpu-developers
+/src/common/snippets/ @openvinotoolkit/openvino-ie-cpu-maintainers
+/src/tests/**cpu** @openvinotoolkit/openvino-ie-cpu-maintainers
 
 # OpenVINO LPT
 /src/common/low_precision_transformations/  @openvinotoolkit/openvino-ie-lpt-maintainers
 
 # OpenVINO GPU:
-/src/inference/include/ie/gpu/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
-/src/inference/include/ie/cldnn/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
```
Comment (ilyachur) [inline CODEOWNERS:49]: Why did you remove these paths?

File: CODEOWNERS [hunk 4]
```diff
@@ -29,64 +29,74 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
-/src/inference/include/openvino/runtime/intel_gpu/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
 /src/plugins/intel_gpu/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
-/docs/snippets/gpu/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
-/docs/OV_Runtime_UG/supported_plugins/GPU.md  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
-/docs/OV_Runtime_UG/supported_plugins/GPU_RemoteTensor_API.md  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
+/src/tests/**gpu** @openvinotoolkit/openvino-ie-gpu-maintainers
 
 # OpenVINO VPU:
 /src/plugins/intel_myriad  @openvinotoolkit/openvino-ie-vpu-maintainers
-/src/inference/include/ie/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers
```
Comment (ilyachur) [inline CODEOWNERS:58]: The same for vpu

File: CODEOWNERS (hunk 2 above)
Comment (AdamTumi) [reply CODEOWNERS:88]: Good catch. I'll fix it.

File: CODEOWNERS (hunk 3 above)
Comment (AdamTumi) [reply CODEOWNERS:49]: These paths are not listed in the reference PPT - that's why I removed them.

File: CODEOWNERS (hunk 4 above)
Comment (AdamTumi) [reply CODEOWNERS:58]: The same as for removed gpu paths.

File: CODEOWNERS (hunk 3 above)
Comment (ilyachur) [reply CODEOWNERS:49]: But we have these paths at the current moment. I don't want to change this logic. It is a legacy gpu part

File: CODEOWNERS (hunk 4 above)
Comment (ilyachur) [reply CODEOWNERS:58]: The same comment as for gpu

Comment (AdamTumi) [reply CODEOWNERS:58]: Reverted this change.

File: CODEOWNERS (hunk 3 above)
Comment (AdamTumi) [reply CODEOWNERS]: Reverted this change.

File: CODEOWNERS [hunk 5]
```diff
@@ -29,64 +29,79 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
-/src/tests_deprecated/unit/engines/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
-/src/tests_deprecated/functional/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
-/src/tests_deprecated/behavior/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
-/src/tests/functional/plugin/myriad/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
-/src/tests/unit/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
-/src/tests/unit/engines/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers @openvinotoolkit/openvino-ie-tests-maintainers
+/src/tests/**myriad** @openvinotoolkit/openvino-ie-vpu-maintainers
```
Comment (ilyachur) [inline CODEOWNERS:66]: I think we also need to add vpu tests

File: CODEOWNERS [hunk 6]
```diff
@@ -29,64 +29,79 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
...
 /src/plugins/intel_gna/  @openvinotoolkit/openvino-ie-gna-maintainers
-/src/inference/include/ie/gna/  @openvinotoolkit/openvino-ie-gna-maintainers
+/src/tests/**gna**  @openvinotoolkit/openvino-ie-gna-maintainers
 
 # OpenVINO ARM CPU:
 /docs/OV_Runtime_UG/supported_plugins/ARM_CPU.md  @openvinotoolkit/openvino_contrib-arm_plugin-maintainers
 
 # OpenVINO Auto (MULTI) plugin:
 /src/plugins/auto/  @openvinotoolkit/openvino-ie-auto-multi-maintainers
-/src/inference/include/ie/multi-device/  @openvinotoolkit/openvino-ie-auto-multi-maintainers
```
Comment (ilyachur) [inline CODEOWNERS:75]: I think we also need to revert this change

**Review by ilyachur — APPROVED** [review]
In general LGTM some minor comments.
And one more common comment.
At the current moment majority of maintainer groups have `-ie-` part, I think we can remove `ie` from group names.

**Pull request merged**